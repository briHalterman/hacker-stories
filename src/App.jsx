import * as React from 'react';

// const SET_STORIES = 'SET_STORIES';
const REMOVE_STORY = 'REMOVE_STORY';

const initialStories = [
  {
    title: 'React',
    url: 'https://reactjs.org/',
    author: 'Jordan Walke',
    num_comments: 3,
    points: 4,
    objectID: 0,
  },
  {
    title: 'Redux',
    url: 'https://redux.js.org/',
    author: 'Dan Abramov, Andrew Clark',
    num_comments: 2,
    points: 5,
    objectID: 1,
  },
];

const getAsyncStories = () =>
  new Promise((resolve) =>
    setTimeout(
      () => resolve({ data: { stories: initialStories } }),
      2000
    )
  );

const storiesReducer = (state, action) => {
  switch (action.type) {
    case 'STORIES_FETCH_INIT':
      return {
        ...state,
        isLoading: true,
        isError: false,
      };
    case 'STORIES_FETCH_SUCCESS':
      return {
        ...state,
        isLoading: false,
        isError: false,
        data: action.payload,
      };
    case 'STORIES_FETCH_FAILURE':
      return {
        ...state,
        isLoading: false,
        isError: true,
      };
    case REMOVE_STORY:
      // return state.filter(
      //   (story) => action.payload.objectID !== story.objectID
      //   );
      return {
        ...state,
        data: state.data.filter(
          (story) => action.payload.objectID !== story.objectID
        ),
      };
    default:
      throw new Error();
  }
};

const useStorageState = (key, initialState) => {
  const [value, setValue] = React.useState(
    localStorage.getItem(key) || initialState
  );

  React.useEffect(() => {
    localStorage.setItem(key, value);
  }, [value, key]);

  return [value, setValue];
};

const App = () => {
  const [searchTerm, setSearchTerm] = useStorageState(
    'search',
    'React'
  );

  const [stories, dispatchStories] = React.useReducer(
    storiesReducer,
    // []
    { data: [], isLoading: false, isError: false }
  );

  // const [isLoading, setIsLoading] = React.useState(false);
  // const [isError, setIsError] = React.useState(false);

  React.useEffect(() => {
    // setIsLoading(true);
    dispatchStories({ type: 'STORIES_FETCH_INIT' });

    getAsyncStories()
      .then((result) => {
        dispatchStories({
          type: 'STORIES_FETCH_SUCCESS',
          payload: result.data.stories,
        });
        // setIsLoading(false);
      })
      // .catch(() => setIsError(true));
      .catch(() =>
        dispatchStories({ type: 'STORIES_FETCH_FAILURE' })
      );
  }, []);

  const handleRemoveStory = (item) => {
    dispatchStories({
      type: REMOVE_STORY,
      payload: item,
    });
  };

  // React.useEffect(() => {
  //   console.log(searchTerm);
  //   localStorage.setItem('search', searchTerm);
  // }, [searchTerm]);

  const handleSearch = (event) => {
    setSearchTerm(event.target.value);

    // localStorage.setItem('search', event.target.value);
  };

  const searchedStories = stories.data.filter((story) =>
    story.title.toLowerCase().includes(searchTerm.toLowerCase())
  );

  return (
    <div>
      <h1>My Hacker Stories</h1>

      <InputWithLabel
        id="search"
        value={searchTerm}
        isFocused
        onInputChange={handleSearch}
      >
        <strong>Search:</strong>
      </InputWithLabel>

      {/* <Search search={searchTerm} onSearch={handleSearch} /> */}

      <hr />

      {stories.isError && <p>Something went wrong ...</p>}

      {stories.isLoading ? (
        <p>Loading ...</p>
      ) : (
        <List
          list={searchedStories}
          onRemoveItem={handleRemoveStory}
        />
      )}
    </div>
  );
};

const InputWithLabel = ({
  id,
  value,
  type = 'text',
  onInputChange,
  isFocused,
  children,
}) => {
  // Create a ref with React's useRef Hook
  const inputRef = React.useRef();

  // Opt into React's lifecycle with React's useEffect Hook
  React.useEffect(() => {
    if (isFocused && inputRef.current) {
      // Execute its focus programmatically as a side-effect, but only if isFocusedis set and the current property is existent
      inputRef.current.focus();
    }
  }, [isFocused]);

  return (
    <>
      <label htmlFor={id}>{children}</label>
      &nbsp;
      {/* Pass ref to element's JSX-reserved ref attribute */}
      <input
        ref={inputRef}
        id={id}
        type={type}
        value={value}
        // autoFocus={isFocused}
        onChange={onInputChange}
      />
    </>
  );
};

// note that `autoFocus` is a shorthand for `autoFocus={true}`
// every attribute that is set to `true` can use this shorthand

// const Search = ({ search, onSearch }) => (
//   <>
//     <label htmlFor="search">Search: </label>
//     <input
//       id="search"
//       type="text"
//       value={search}
//       onChange={onSearch}
//     />
//   </>
// );

const List = ({ list, onRemoveItem }) => (
  <ul>
    {list.map((item) => (
      <Item
        key={item.objectID}
        item={item}
        onRemoveItem={onRemoveItem}
      />
    ))}
  </ul>
);

const Item = ({ item, onRemoveItem }) => (
  <li>
    <span>
      <a href={item.url}>{item.title}</a>
    </span>
    <span>{item.author}</span>
    <span>{item.num_comments}</span>
    <span>{item.points}</span>
    <span>
      <button type="button" onClick={() => onRemoveItem(item)}>
        Dismiss
      </button>
    </span>
  </li>
);

export default App;
