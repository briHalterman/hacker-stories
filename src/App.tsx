// import clsx from 'clsx';
import * as React from 'react';
import axios from 'axios';
import styled, { createGlobalStyle } from 'styled-components';
// import styles from './App.module.css';
import { List } from './List';

import { FaHackerNews } from 'react-icons/fa';
// import { AiOutlineCheck } from 'react-icons/ai';

import InputWithLabel from './InputWithLabel';

type Story = {
  objectID: string;
  url: string;
  title: string;
  author: string;
  num_comments: number;
  points: number;
};

type Stories = Story[];

type StoriesState = {
  data: Stories;
  isLoading: boolean;
  isError: boolean;
};

type StoriesFetchInitAction = {
  type: 'STORIES_FETCH_INIT';
};

type StoriesFetchSuccessAction = {
  type: 'STORIES_FETCH_SUCCESS';
  payload: Stories;
};

type StoriesFetchFailureAction = {
  type: 'STORIES_FETCH_FAILURE';
};

type StoriesRemoveAction = {
  type: 'REMOVE_STORY';
  payload: Story;
};

type StoriesAction =
  | StoriesFetchInitAction
  | StoriesFetchSuccessAction
  | StoriesFetchFailureAction
  | StoriesRemoveAction;

const REMOVE_STORY = 'REMOVE_STORY';
const STORIES_FETCH_INIT = 'STORIES_FETCH_INIT';
const STORIES_FETCH_SUCCESS = 'STORIES_FETCH_SUCCESS';
const STORIES_FETCH_FAILURE = 'STORIES_FETCH_FAILURE';

const storiesReducer = (
  state: StoriesState,
  action: StoriesAction
) => {
  switch (action.type) {
    case STORIES_FETCH_INIT:
      return {
        ...state,
        isLoading: true,
        isError: false,
      };
    case STORIES_FETCH_SUCCESS:
      return {
        ...state,
        isLoading: false,
        isError: false,
        data: action.payload,
      };
    case STORIES_FETCH_FAILURE:
      return {
        ...state,
        isLoading: false,
        isError: true,
      };
    case REMOVE_STORY:
      return {
        ...state,
        data: state.data.filter(
          (story) => action.payload.objectID !== story.objectID
        ),
      };
    default:
      throw new Error();
  }
};

const useStorageState = (
  key: string,
  initialState: string
): [string, (newValue: string) => void] => {
  const [value, setValue] = React.useState(
    localStorage.getItem(key) || initialState
  );

  React.useEffect(() => {
    localStorage.setItem(key, value);
  }, [value, key]);

  return [value, setValue];
};

// API_ENDPOINT used to fetch popular tech stories for a certain query
const API_ENDPOINT = 'https://hn.algolia.com/api/v1/search?query=';

const GlobalStyle = createGlobalStyle`
  body {
    margin: 0;
    font-family: system-ui, -apple-system, BlinkMacSystemFont, 'Segoe UI', Roboto, Oxygen, Ubuntu, Cantarell, 'Open Sans', 'Helvetica Neue', sans-serif, BlinkMacSystemFont, 'Segoe UI', 'Roboto', 'Oxygen', 'Ubantu', 'Cantarell', 'Fira Sans', 'Droid Sans', 'Helvetica Neue', sans-serif;
    -webkit-font-smoothing: antialiased;
    -moz-osx-font-smoothing: grayscale;
  }
`;

const StyledContainer = styled.div`
  height: 100vw;
  padding: 20px;

  background: #83a4d4;
  background: linear-gradient(to left, #b6fbff, #83a4d4);

  color: #171212;
`;

const StyledHeadlinePrimary = styled.h1`
  font-size: 48px;
  font-weight: 300;
  letter-spacing: 2px;
`;

const StyledButton = styled.button`
  background: transparent;
  border: 1px solid #171212;
  padding: 5px;
  cursor: pointer;

  transition: all 0.1s ease-in;

  &:hover {
    background: #171212;
    color: #ffffff;
  }
`;

const StyledButtonLarge = styled(StyledButton)`
  padding: 10px;
`;

const StyledSearchForm = styled.form`
  padding: 10px 0 20px 0;
  display: flex;
  align-items: baseline;
`;

const App = () => {
  const [searchTerm, setSearchTerm] = useStorageState(
    'search',
    'React'
  );

  const [url, setUrl] = React.useState(
    `${API_ENDPOINT}${searchTerm}`
  );

  const [stories, dispatchStories] = React.useReducer(
    storiesReducer,
    { data: [], isLoading: false, isError: false }
  );

  // Move all data fetching logic from the side-effect into arrow function expression
  const handleFetchStories = React.useCallback(async () => {
    // wrap function into React's useCallback hook
    //  if `searchTerm is not present do nothing

    dispatchStories({ type: 'STORIES_FETCH_INIT' });

    try {
      const result = await axios.get(url);

      dispatchStories({
        type: 'STORIES_FETCH_SUCCESS',
        payload: result.data.hits, // Send returned result (different data structure) as payload to component's state reducer
      });
    } catch {
      dispatchStories({ type: 'STORIES_FETCH_FAILURE' });
    }
  }, [url]);

  React.useEffect(() => {
    handleFetchStories(); // Invoke handleFetchStories function in useEffect Hook
  }, [handleFetchStories]);

  const handleRemoveStory = (item: Story) => {
    dispatchStories({
      type: REMOVE_STORY,
      payload: item,
    });
  };

  const handleSearchInput = (
    event: React.ChangeEvent<HTMLInputElement>
  ) => {
    setSearchTerm(event.target.value);
  };

  const handleSearchSubmit = (
    event: React.FormEvent<HTMLFormElement>
  ) => {
    setUrl(`${API_ENDPOINT}${searchTerm}`);

    event.preventDefault();
  };

  return (
    <>
      <GlobalStyle />

      <StyledContainer>
        <StyledHeadlinePrimary>
          <FaHackerNews size="50" />
          My Hacker Stories
        </StyledHeadlinePrimary>

        <SearchForm
          searchTerm={searchTerm}
          onSearchInput={handleSearchInput}
          onSearchSubmit={handleSearchSubmit}
          // className="button_large"
        />

        {/* <hr /> */}

        {stories.isError && <p>Something went wrong ...</p>}

        {stories.isLoading ? (
          <p>Loading ...</p>
        ) : (
          <List
            list={stories.data}
            onRemoveItem={handleRemoveStory}
          />
        )}
      </StyledContainer>
    </>
  );
};

type SearchFormProps = {
  searchTerm: string;
  onSearchInput: (event: React.ChangeEvent<HTMLInputElement>) => void;
  onSearchSubmit: (event: React.FormEvent<HTMLFormElement>) => void;
};

const SearchForm: React.FC<SearchFormProps> = ({
  searchTerm,
  onSearchInput,
  onSearchSubmit,
  // className,
}) => (
  <StyledSearchForm onSubmit={onSearchSubmit}>
    <InputWithLabel
      id="search"
      value={searchTerm}
      isFocused
      onInputChange={onSearchInput}
    >
      {/* <strong>Search:</strong> */}
      Search:
    </InputWithLabel>

    <StyledButtonLarge
      type="submit"
      disabled={!searchTerm}
      // className={clsx(styles.button, styles.buttonLarge)}
      // className={clsx(styles.button, {
      //   [styles.buttonLarge]: isLarge,
      // })}
    >
      Submit
    </StyledButtonLarge>
  </StyledSearchForm>
);

export default App;

export { storiesReducer, SearchForm, InputWithLabel, List };
