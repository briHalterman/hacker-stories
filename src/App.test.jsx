// describe('something truthy and falsy', () => {
//   it('true to be true', () => {
//     expect(true).toBeTruthy();
//   });

//   it('false to be false', () => {
//     expect(false).toBeFalsy();
//   });
// });

import { describe, it, expect, vi } from 'vitest';
import {
  render,
  screen,
  fireEvent,
  waitFor,
} from '@testing-library/react';

import App, {
  storiesReducer,
  Item,
  List,
  SearchForm,
  InputWithLabel,
} from './App';

const initialState = {
  data: [],
  isLoading: false,
  isError: false,
};

const storyOne = {
  title: 'React',
  url: 'https://reactjs.org/',
  author: 'Jordan Walke',
  num_comments: 3,
  points: 4,
  objectID: 0,
};

const storyTwo = {
  title: 'Redux',
  url: 'https://redux.js.org/',
  author: 'Dan Abramov, Andrew Clark',
  num_comments: 2,
  points: 5,
  objectID: 1,
};

const stories = [storyOne, storyTwo];

describe('storiesReducer', () => {
  it('fetches all stories', () => {
    const action = { type: 'STORIES_FETCH_INIT' };
    const newState = storiesReducer(initialState, action);

    expect(newState.isLoading).toBeTruthy();
    expect(newState.isError).toBeFalsy();
  });

  it('correctly updates application state', () => {
    const action = {
      type: 'STORIES_FETCH_SUCCESS',
      payload: stories,
    };
    const newState = storiesReducer(initialState, action);

    expect(newState.isLoading).toBeFalsy();
    expect(newState.isError).toBeFalsy();
    expect(newState.data).toHaveLength(stories.length);
    expect(newState.data).toEqual(stories);
  });

  it('handles failed API request', () => {
    const action = { type: 'STORIES_FETCH_FAILURE' };
    const newState = storiesReducer(initialState, action);

    expect(newState.isLoading).toBeFalsy();
    expect(newState.isError).toBeTruthy();
  });

  it('removes a story from all stories', () => {
    const action = { type: 'REMOVE_STORY', payload: storyOne };
    const state = { data: stories, isLoading: false, isError: false };

    const newState = storiesReducer(state, action);

    const expectedState = {
      data: [storyTwo],
      isLoading: false,
      isError: false,
    };

    expect(newState).toStrictEqual(expectedState);
  });

  describe('Item', () => {
    it('renders all properties', () => {
      render(<Item item={storyOne} />);

      expect(screen.getByText('Jordan Walke')).toBeInTheDocument();
      expect(screen.getByText('React')).toHaveAttribute(
        'href',
        'https://reactjs.org/'
      );
    });

    it('renders a clickable dismiss button', () => {
      render(<Item item={storyOne} />);

      // screen.getByRole('');

      expect(screen.getByRole('button')).toBeInTheDocument();
    });

    it('clicking the dismiss button calls the callback handler', () => {
      const handleRemoveItem = vi.fn();

      render(
        <Item item={storyOne} onRemoveItem={handleRemoveItem} />
      );

      fireEvent.click(screen.getByRole('button'));

      expect(handleRemoveItem).toHaveBeenCalledTimes(1);
      expect(handleRemoveItem).toHaveBeenCalledWith(storyOne);
    });
  });
});

describe('SearchForm', () => {
  const searchFormProps = {
    searchTerm: 'React',
    onSearchInput: vi.fn(),
    onSearchSubmit: vi.fn(),
  };

  it('renders the input field with its value', () => {
    render(<SearchForm {...searchFormProps} />);

    // screen.debug();
    expect(screen.getByDisplayValue('React')).toBeInTheDocument();
  });

  it('renders the correct label', () => {
    render(<SearchForm {...searchFormProps} />);

    expect(screen.getByLabelText(/Search/)).toBeInTheDocument();
  });

  it('calls onSearchInput on input field change', () => {
    render(<SearchForm {...searchFormProps} />);

    const inputElement = screen.getByDisplayValue('React');

    fireEvent.change(inputElement, {
      target: { value: 'Redux' },
    });

    expect(searchFormProps.onSearchInput).toHaveBeenCalledTimes(1);

    expect(searchFormProps.onSearchInput).toHaveBeenCalledWith(
      expect.any(Object)
    );
  });

  it('calls onSearchSubmit on button submit click', () => {
    render(<SearchForm {...searchFormProps} />);

    fireEvent.submit(screen.getByRole('button'));

    expect(searchFormProps.onSearchSubmit).toHaveBeenCalledTimes(1);
    expect(searchFormProps.onSearchSubmit).toHaveBeenCalledWith(
      expect.any(Object)
    );
  });
});

describe('List', () => {
  const handleRemoveItem = vi.fn();

  it('renders the correct number of items', () => {
    render(<List list={stories} onRemoveItem={handleRemoveItem} />);

    expect(screen.getAllByRole('listitem')).toHaveLength(
      stories.length
    );
  });

  it('calls onRemove when the button is clicked', () => {
    render(<List list={stories} onRemoveItem={handleRemoveItem} />);

    const removeButtons = screen.getAllByRole('button');

    fireEvent.click(removeButtons[0]);

    expect(handleRemoveItem).toHaveBeenCalledTimes(1);
    expect(handleRemoveItem).toHaveBeenCalledWith(stories[0]);
  });
});

describe('InputWithLabel', () => {
  const handleChange = vi.fn();
  const labelText = 'Search';

  it('renders the label correctly', () => {
    render(
      <InputWithLabel
        id="search"
        value="React"
        onInputChange={handleChange}
      >
        {labelText}
      </InputWithLabel>
    );

    expect(screen.getByLabelText(labelText)).toBeInTheDocument();
  });

  it('renders the input field with the correct value', () => {
    render(
      <InputWithLabel
        id="search"
        value="React"
        onInputChange={handleChange}
      >
        {labelText}
      </InputWithLabel>
    );

    expect(screen.getByDisplayValue('React')).toBeInTheDocument();
  });

  it('calls onInputChange when text is entered', () => {
    render(
      <InputWithLabel
        id="search"
        value=""
        onInputChange={handleChange}
      >
        {labelText}
      </InputWithLabel>
    );

    const input = screen.getByLabelText(labelText);
    fireEvent.change(input, { target: { value: 'Redux' } });

    expect(handleChange).toHaveBeenCalledTimes(1);
  });
});
