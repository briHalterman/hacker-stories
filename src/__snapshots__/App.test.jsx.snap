// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`App > renders snapshot 1`] = `
<div
  class="sc-jMLmsk cxLMdr"
>
  <h1
    class="sc-cMHOsR gxjCor"
  >
    <svg
      fill="currentColor"
      height="50"
      stroke="currentColor"
      stroke-width="0"
      viewBox="0 0 448 512"
      width="50"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M0 32v448h448V32H0zm21.2 197.2H21c.1-.1.2-.3.3-.4 0 .1 0 .3-.1.4zm218 53.9V384h-31.4V281.3L128 128h37.3c52.5 98.3 49.2 101.2 59.3 125.6 12.3-27 5.8-24.4 60.6-125.6H320l-80.8 155.1z"
      />
    </svg>
    My Hacker Stories
  </h1>
  <form
    class="sc-cZztgT gywJLP"
  >
    <label
      class="sc-hWWBcw gPqGSH"
      for="search"
    >
      Search:
    </label>
     
    <input
      class="sc-jHbxoU ewemwe"
      id="search"
      type="text"
      value="JavaScript"
    />
    <button
      class="sc-cGXZpB sc-jZCRgm dGHqHG ikJyLe"
      type="submit"
    >
      Submit
    </button>
  </form>
  <button
    type="button"
  >
    JavaScript
  </button>
  <ul>
    <div
      class="sc-eaUbBy dvkHrQ"
    >
      <span
        class="sc-gaZyOd hA-DaVc"
        width="40%"
      >
        <button
          class="sc-dDvxFM sc-gTrWKq ga-Dddz hpxlC"
        >
          Title 
        </button>
      </span>
      <span
        class="sc-gaZyOd jKrUdn"
        width="30%"
      >
        <button
          class="sc-dDvxFM sc-gTrWKq ga-Dddz hpxlC"
        >
          Author 
        </button>
      </span>
      <span
        class="sc-gaZyOd ggsWWl"
        width="10%"
      >
        <button
          class="sc-dDvxFM sc-gTrWKq ga-Dddz hpxlC"
        >
          Comments 
        </button>
      </span>
      <span
        class="sc-gaZyOd ggsWWl"
        width="10%"
      >
        <button
          class="sc-dDvxFM sc-gTrWKq ga-Dddz hpxlC"
        >
          Points 
        </button>
      </span>
      <span
        class="sc-gaZyOd ggsWWl"
        width="10%"
      >
        Actions
      </span>
    </div>
    <li
      class="sc-KXCwU biGGsF"
    >
      <span
        class="sc-gyycJP eNgHDH"
        width="40%"
      >
        <a
          href="https://reactjs.org/"
        >
          React
        </a>
      </span>
      <span
        class="sc-gyycJP fELikQ"
        width="30%"
      >
        Jordan Walke
      </span>
      <span
        class="sc-gyycJP iSAiuG"
        width="10%"
      >
        3
      </span>
      <span
        class="sc-gyycJP iSAiuG"
        width="10%"
      >
        4
      </span>
      <span
        class="sc-gyycJP iSAiuG"
        width="10%"
      >
        <button
          class="sc-dDvxFM sc-jqVXSH ga-Dddz gRqVEr"
          type="button"
        >
          Dismiss
           
          <svg
            color="green"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            style="color: green;"
            viewBox="0 0 1024 1024"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M912 190h-69.9c-9.8 0-19.1 4.5-25.1 12.2L404.7 724.5 207 474a32 32 0 0 0-25.1-12.2H112c-6.7 0-10.4 7.7-6.3 12.9l273.9 347c12.8 16.2 37.4 16.2 50.3 0l488.4-618.9c4.1-5.1.4-12.8-6.3-12.8z"
            />
          </svg>
        </button>
      </span>
    </li>
    <li
      class="sc-KXCwU biGGsF"
    >
      <span
        class="sc-gyycJP eNgHDH"
        width="40%"
      >
        <a
          href="https://redux.js.org/"
        >
          Redux
        </a>
      </span>
      <span
        class="sc-gyycJP fELikQ"
        width="30%"
      >
        Dan Abramov, Andrew Clark
      </span>
      <span
        class="sc-gyycJP iSAiuG"
        width="10%"
      >
        2
      </span>
      <span
        class="sc-gyycJP iSAiuG"
        width="10%"
      >
        5
      </span>
      <span
        class="sc-gyycJP iSAiuG"
        width="10%"
      >
        <button
          class="sc-dDvxFM sc-jqVXSH ga-Dddz gRqVEr"
          type="button"
        >
          Dismiss
           
          <svg
            color="green"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            style="color: green;"
            viewBox="0 0 1024 1024"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M912 190h-69.9c-9.8 0-19.1 4.5-25.1 12.2L404.7 724.5 207 474a32 32 0 0 0-25.1-12.2H112c-6.7 0-10.4 7.7-6.3 12.9l273.9 347c12.8 16.2 37.4 16.2 50.3 0l488.4-618.9c4.1-5.1.4-12.8-6.3-12.8z"
            />
          </svg>
        </button>
      </span>
    </li>
  </ul>
</div>
`;

exports[`InputWithLabel > renders snapshot 1`] = `
<label
  class="sc-hWWBcw gPqGSH"
  for="search"
>
  Search
</label>
`;

exports[`List > renders snapshot 1`] = `
<ul>
  <div
    class="sc-eaUbBy dvkHrQ"
  >
    <span
      class="sc-gaZyOd hA-DaVc"
      width="40%"
    >
      <button
        class="sc-dDvxFM sc-gTrWKq ga-Dddz hpxlC"
      >
        Title 
      </button>
    </span>
    <span
      class="sc-gaZyOd jKrUdn"
      width="30%"
    >
      <button
        class="sc-dDvxFM sc-gTrWKq ga-Dddz hpxlC"
      >
        Author 
      </button>
    </span>
    <span
      class="sc-gaZyOd ggsWWl"
      width="10%"
    >
      <button
        class="sc-dDvxFM sc-gTrWKq ga-Dddz hpxlC"
      >
        Comments 
      </button>
    </span>
    <span
      class="sc-gaZyOd ggsWWl"
      width="10%"
    >
      <button
        class="sc-dDvxFM sc-gTrWKq ga-Dddz hpxlC"
      >
        Points 
      </button>
    </span>
    <span
      class="sc-gaZyOd ggsWWl"
      width="10%"
    >
      Actions
    </span>
  </div>
  <li
    class="sc-KXCwU biGGsF"
  >
    <span
      class="sc-gyycJP eNgHDH"
      width="40%"
    >
      <a
        href="https://reactjs.org/"
      >
        React
      </a>
    </span>
    <span
      class="sc-gyycJP fELikQ"
      width="30%"
    >
      Jordan Walke
    </span>
    <span
      class="sc-gyycJP iSAiuG"
      width="10%"
    >
      3
    </span>
    <span
      class="sc-gyycJP iSAiuG"
      width="10%"
    >
      4
    </span>
    <span
      class="sc-gyycJP iSAiuG"
      width="10%"
    >
      <button
        class="sc-dDvxFM sc-jqVXSH ga-Dddz gRqVEr"
        type="button"
      >
        Dismiss
         
        <svg
          color="green"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          style="color: green;"
          viewBox="0 0 1024 1024"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M912 190h-69.9c-9.8 0-19.1 4.5-25.1 12.2L404.7 724.5 207 474a32 32 0 0 0-25.1-12.2H112c-6.7 0-10.4 7.7-6.3 12.9l273.9 347c12.8 16.2 37.4 16.2 50.3 0l488.4-618.9c4.1-5.1.4-12.8-6.3-12.8z"
          />
        </svg>
      </button>
    </span>
  </li>
  <li
    class="sc-KXCwU biGGsF"
  >
    <span
      class="sc-gyycJP eNgHDH"
      width="40%"
    >
      <a
        href="https://redux.js.org/"
      >
        Redux
      </a>
    </span>
    <span
      class="sc-gyycJP fELikQ"
      width="30%"
    >
      Dan Abramov, Andrew Clark
    </span>
    <span
      class="sc-gyycJP iSAiuG"
      width="10%"
    >
      2
    </span>
    <span
      class="sc-gyycJP iSAiuG"
      width="10%"
    >
      5
    </span>
    <span
      class="sc-gyycJP iSAiuG"
      width="10%"
    >
      <button
        class="sc-dDvxFM sc-jqVXSH ga-Dddz gRqVEr"
        type="button"
      >
        Dismiss
         
        <svg
          color="green"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          style="color: green;"
          viewBox="0 0 1024 1024"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M912 190h-69.9c-9.8 0-19.1 4.5-25.1 12.2L404.7 724.5 207 474a32 32 0 0 0-25.1-12.2H112c-6.7 0-10.4 7.7-6.3 12.9l273.9 347c12.8 16.2 37.4 16.2 50.3 0l488.4-618.9c4.1-5.1.4-12.8-6.3-12.8z"
          />
        </svg>
      </button>
    </span>
  </li>
</ul>
`;

exports[`SearchForm > renders snapshot 1`] = `
<form
  class="sc-cZztgT gywJLP"
>
  <label
    class="sc-hWWBcw gPqGSH"
    for="search"
  >
    Search:
  </label>
   
  <input
    class="sc-jHbxoU ewemwe"
    id="search"
    type="text"
    value="React"
  />
  <button
    class="sc-cGXZpB sc-jZCRgm dGHqHG ikJyLe"
    type="submit"
  >
    Submit
  </button>
</form>
`;
